virtual-classroom/src/main/java/ro/utcn/sd/mid/assign1/virtualclassroom/controller/ConsoleController.java

```diff
@@ -0,0 +1,360 @@
+        print("Available tags: ");
+        tagService.listTags().forEach(tag -> print(tag.getTagName() + " "));
+        //println("");
+        while (!answered) {
+            println("Do you wish to add any tags to your question? [yes/no]");
+            String yesNo = scanner.next().trim().toLowerCase();
+            if (yesNo.indexOf('y') >= 0) {
+                println("Specify the tags by separating them with a space \" \". " +
+                        "The tags that are not existent will be created and assigned " +
+                        "to your question.");
+                print("Tags to be added: ");
+                String tagLine = scanner.next();
+                String[] tags = tagLine.split(" ");
+                for (String tagString : tags) {
+                    //println("tag: " + tagString);
+                    try {
+                        Tag retrievedTag = tagService.findTagByTagName(tagString);
+                        questionService.addTagToQuestion(retrievedTag, questionToBeCreated);
+                    } catch (TagNotFoundException e) {
+                        questionService.addTagToQuestion(tagService.
+                                createTag(new Tag(tagString)), questionToBeCreated);
+                    }
+                }
+                questionService.askQuestion(questionToBeCreated);
+                println("Question " + questionToBeCreated.getId() + " was created.");
+                answered = true;
+            } else {
+                if (yesNo.indexOf('n') >= 0) {
+                    questionService.askQuestion(questionToBeCreated);
+                    println("Your question was created.");
+                    answered = true;
+                } else
+                    println("Invalid command.");
+            }
+        }
+    }
+
+
+    // ================== Feature 2 handlers ==================
+
+
+    @SuppressWarnings("Duplicates")
+    private void handleListOne() {
+        print("Id of the question to list: ");
+        String id = scanner.next();
+        try {
+            int qId = Integer.parseInt(id);
+            try {
+                Question question = questionService.findById(qId);
+                println(question.toString());
+                answerService.listAnswersForQuestion(question).forEach(
+                        answer -> println("   " + answer.toString())
+                );
+            } catch (QuestionNotFoundException e) {
+                println("There is no question with id = " + qId);
+            }
+        } catch (NumberFormatException e) {
+            println("Invalid id. Please introduce a number.");
+        }
+    }
+
+    @SuppressWarnings("Duplicates")
+    private void handleAnswer() {
+        print("Id of the question to answer: ");
+        String id = scanner.next();
+        try {
+            int qId = Integer.parseInt(id);
+            try {
+                Question question = questionService.findById(qId);
```

Comment:
You have some business logic inside the controller. You should simply collect the user input (question ID, answer text) and pass this to the service layer. The service layer should then load the question, create the answer and save it.